src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/MetadataEETypeNode.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+
+using Internal.Runtime;
+using Internal.TypeSystem;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    public sealed class MetadataEETypeNode : EETypeNode
+    {
+        public MetadataEETypeNode(NodeFactory factory, TypeDesc type) : base(factory, type)
+        {
+            Debug.Assert(!type.IsCanonicalDefinitionType(CanonicalFormKind.Any));
+        }
+
+        protected override bool IsReflectionVisible => true;
+
+        protected override string GetName(NodeFactory factory) => this.GetMangledName(factory.NameMangler) + " with metadata";
+
+        public override bool ShouldSkipEmittingObjectNode(NodeFactory factory)
+        {
+            // If there is a constructed version of this node in the graph, emit that instead
+            if (ConstructedEETypeNode.CreationAllowed(_type))
+                return factory.ConstructedTypeSymbol(_type).Marked;
+
+            return false;
+        }
+
+        protected override DependencyList ComputeNonRelocationBasedDependencies(NodeFactory factory)
+        {
+            DependencyList dependencyList = base.ComputeNonRelocationBasedDependencies(factory);
+
+            // Ensure that we track the necessary type symbol if we are working with a metadata type symbol.
+            // The emitter will ensure we don't emit both, but this allows us assert that we only generate
+            // relocs to nodes we emit.
+            dependencyList.Add(factory.NecessaryTypeSymbol(_type), "NecessaryType for metadata type");
+
+            if (_type is MetadataType mdType)
+                ModuleUseBasedDependencyAlgorithm.AddDependenciesDueToModuleUse(ref dependencyList, factory, mdType.Module);
+
+            // Ask the metadata manager if we have any dependencies due to the presence of the EEType.
+            factory.MetadataManager.GetDependenciesDueToEETypePresence(ref dependencyList, factory, _type);
```

Comment:
Let's remove the call to this method in `ConstructedEETypeNode` now that we're adding these dependencies here.

Reply:
Done